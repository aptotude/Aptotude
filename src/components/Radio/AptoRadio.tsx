import './aptoRadio.scss';
export { Radio as AptoRadio } from 'semantic-ui-react';
