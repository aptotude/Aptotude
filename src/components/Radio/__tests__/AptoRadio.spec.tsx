import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoRadio } from '../AptoRadio';

afterEach(() => {
  cleanup();
});

describe('AptoRadio', () => {
  it('Renders Component', () => {
    const { container } = render(<AptoRadio />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders with name/value', () => {
    const { container } = render(<AptoRadio name="foo" value="bar" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders checked', () => {
    const { container } = render(
      <AptoRadio checked={true} name="foo" value="bar" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders disabled', () => {
    const { container } = render(
      <AptoRadio disabled={true} name="foo" value="bar" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
