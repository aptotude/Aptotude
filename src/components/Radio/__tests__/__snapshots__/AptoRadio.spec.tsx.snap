// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoRadio Renders Component 1`] = `
<div
  class="ui fitted radio checkbox"
>
  <input
    class="hidden"
    readonly=""
    tabindex="0"
    type="radio"
    value=""
  />
  <label />
</div>
`;

exports[`AptoRadio Renders checked 1`] = `
<div
  class="ui checked fitted radio checkbox"
>
  <input
    checked=""
    class="hidden"
    name="foo"
    readonly=""
    tabindex="0"
    type="radio"
    value="bar"
  />
  <label />
</div>
`;

exports[`AptoRadio Renders disabled 1`] = `
<div
  class="ui disabled fitted radio checkbox"
>
  <input
    class="hidden"
    disabled=""
    name="foo"
    readonly=""
    tabindex="-1"
    type="radio"
    value="bar"
  />
  <label />
</div>
`;

exports[`AptoRadio Renders with name/value 1`] = `
<div
  class="ui fitted radio checkbox"
>
  <input
    class="hidden"
    name="foo"
    readonly=""
    tabindex="0"
    type="radio"
    value="bar"
  />
  <label />
</div>
`;
